### DISCUSSION TITLE
GetAttribLocation not working.

### DISCUSSION THREAD

**Discussion by Hromon** [dotnet/Silk.NET#550] [Q&A] [answered]
# Summary
GetAttribLocation always return -1.

# Steps to reproduce
- Platform: Desktop
- Framework Version: .NET 5
- API: OpenGL
- API Version: OpenGL 3.3 Core

![image](https://user-images.githubusercontent.com/29508584/126530307-48e0841c-b898-40ff-b669-9cd9526fc5ba.png)
![image](https://user-images.githubusercontent.com/29508584/126530389-690ea230-2ff8-4e37-a60c-a2fbe65768d4.png)
![image](https://user-images.githubusercontent.com/29508584/126530410-86c1445b-839b-4cd7-9555-c67c4ce91e62.png)

**Comment by Perksey** [comment, marked as answer]
That's because `Ir` is not an attribute. It is a uniform, and you'll need to use `GetUniformLocation` for uniforms.

<sub>Note that the Silk.NET issue tracker is for Silk.NET issues only, general support & external issues live in the Discussions tab. If you are unsure whether a bug or issue is Silk.NET's fault, external issue, or misuse; open a Discussion and if we determine it's a Silk.NET issue we will promote it to an Issue as we see fit.</sub>

**Reply by Hromon** [reply]
![image](https://user-images.githubusercontent.com/29508584/126656740-82c7fdd3-7d7e-439c-83d9-6c123b0db973.png)
![image](https://user-images.githubusercontent.com/29508584/126656767-7b93b7a1-798e-41ef-a2d3-fd0bfec42f10.png)
![image](https://user-images.githubusercontent.com/29508584/126656795-11139a1b-4669-4707-869f-69f19784b18b.png)
![image](https://user-images.githubusercontent.com/29508584/126656829-d51a2db8-3d6d-4044-953e-ba1b85b14cb0.png)
